spec.html

```diff
@@ -5546,6 +5746,110 @@ <h1>CopyDataProperties ( _target_, _source_, _excludedItems_ )</h1>
       </emu-note>
     </emu-clause>
+
+    <emu-clause id="sec-newprivatename" aoid="NewPrivateName">
+      <h1>NewPrivateName ( _description_ )</h1>
+      <p>The abstract operation NewPrivateName takes argument _description_ (a String). It performs the following steps when called:</p>
+      <emu-alg>
+        1. Assert: Type(_description_) is String.
+        1. Return a new unique Private Name whose [[Description]] value is _description_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privateelementfind" aoid="PrivateElementFind">
+      <h1>PrivateElementFind ( _P_, _O_ )</h1>
+      <p>The abstract operation PrivateElementFind takes arguments _P_ (a Private Name) and _O_ (an Object). It performs the following steps when called:</p>
+      <emu-alg>
+        1. For each PrivateElement Record _entry_ of _O_.[[PrivateElements]], do
+          1. If _entry_.[[Key]] is _P_, return _entry_.
+        1. Return ~empty~.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privatefieldadd" aoid="PrivateFieldAdd">
+      <h1>PrivateFieldAdd ( _P_, _O_, _value_ )</h1>
+      <p>The abstract operation PrivateFieldAdd takes arguments _P_ (a Private Name), _O_ (an Object), and _value_ (an ECMAScript language value). It performs the following steps when called:</p>
+      <emu-alg>
+        1. Let _entry_ be ! PrivateElementFind(_P_, _O_).
+        1. If _entry_ is not ~empty~, throw a *TypeError* exception.
+        1. Append PrivateElement { [[Key]]: _P_, [[Kind]]: ~field~, [[Value]]: _value_ } to _O_.[[PrivateElements]].
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privatemethodadd" aoid="PrivateMethodAdd">
+      <h1>PrivateMethodAdd ( _method_, _O_ )</h1>
+      <p>The abstract operation PrivateMethodAdd takes arguments _method_ (a PrivateElement Record) and _O_ (an Object). It performs the following steps when called:</p>
+      <emu-alg>
+        1. Assert: _method_.[[Kind]] is either ~method~ or ~accessor~.
+        1. Let _entry_ be ! PrivateElementFind(_method_.[[Key]], _O_).
+        1. If _entry_ is not ~empty~, throw a *TypeError* exception.
+        1. Append _method_ to _O_.[[PrivateElements]].
+        1. NOTE: the values for private methods and accessors are shared across instances. This step does not create a new copy of the method or accessor.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privateget" aoid="PrivateGet">
+      <h1>PrivateGet ( _P_, _O_ )</h1>
+      <p>The abstract operation PrivateGet takes arguments _P_ (a Private Name) and _O_ (an Object). It performs the following steps when called:</p>
+      <emu-alg>
+        1. Let _entry_ be ! PrivateElementFind(_P_, _O_).
+        1. If _entry_ is ~empty~, throw a *TypeError* exception.
+        1. If _entry_.[[Kind]] is ~field~ or ~method~, then
+          1. Return _entry_.[[Value]].
+        1. Assert: _P_.[[Kind]] is ~accessor~.
+        1. If _P_.[[Get]] is *undefined*, throw a *TypeError* exception.
```

Comment:
That's been the case for [the entire lifetime of the proposal](https://github.com/tc39/proposal-private-methods/blob/30f3bcf9c612088b516270e5c38ca5ad27650470/spec.html#L643), through a number of revisions and nearly four years of discussion. I'd certainly hope it was noticed by reviewers. I noticed, at any rate.

These semantics are what we got consensus on, and are now web reality, so if you'd like to propose different behavior a separate proposal is probably in order.

Reply:
Fair enough. It could easily change since it's currently an error, but maybe there's no use case for it to do so.

Reply:
IIRC, this is intentional based on the same design decision we made for throwing `TypeError` when a private field is not installed. I remember discussing in plenary about making this `SyntaxError` instead, since we already know at compile time that we are trying to apply `[[Get]]` for a not defined private getter, but we decided to keep current semantics. My memory might be tricking me, but I think the reason to throw `TypeError` is to make this proposal more friendly to a future decorators proposal.